node.go

```diff
@@ -56,6 +56,9 @@ func RemoveAllTagsOfNode() NodeOption {
 
 // WithResourceOfNode will add or overwrite the amount of a target Resource the node provides
 func WithResourceOfNode(targetResource Resource, resourceAmount int64) NodeOption {
+	if resourceAmount < 0 {
+		panic("resourceAmount cannot be negative")
+	}
```

Comment:
Fail fast on all of these annoying `< 0` checks, can't do much with them so might as well alert the caller ASAP. One hacker-mindset way of doing this is to change all fields to be `uint`, but then we're making the API obscure and losing any concrete error checking for these, also not 100% sure how the conversion from a negative integer would look like to a `uint`, might be machine/platform/sys dependent.